app/src/main/java/com/bogdan/codeforceswatcher/activity/MainActivity.kt

```diff
@@ -44,15 +52,30 @@ class MainActivity : AppCompatActivity(), SwipeRefreshLayout.OnRefreshListener,
 
     override fun onCreateOptionsMenu(menu: Menu): Boolean {
         menuInflater.inflate(R.menu.menu_main, menu)
+        showItemMenu(menu.findItem(R.id.action_descending_ascending), true)
         return true
     }
 
+    @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        when (item.itemId) {
+            R.id.action_descending_ascending -> {
+                showItemMenu(item, false)
+                counterIcon = (counterIcon + 1) % 3
+                val sharedPrefs = getPreferences(MODE_PRIVATE)
```

Comment:
Please, move all `SharedPreferences` operations to `Prefs` `object`. Here is an example (in Java):
```java
package com.ekanivets.lyrsenseradio.utils;

import android.content.Context;
import android.content.SharedPreferences;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import com.ekanivets.lyrsenseradio.LrApp;
import com.ekanivets.lyrsenseradio.model.Sites;

/**
 * Util class to encapsulate Shared Preferences handling logic.
 * Created 2/24/17.
 *
 * @author Evgenii Kanivets
 */

public class Prefs {

    private static final String KEY_SITES = "key_sites";
    private static final String KEY_COOKIE = "key_cookie";
    private static final String KEY_INTRO_SHOWN = "key_intro_shown";

    private static Prefs prefs;

    public static Prefs get() {
        if (prefs == null) {
            prefs = new Prefs(LrApp.get());
        }
        return prefs;
    }

    @NonNull
    private Context context;

    private Prefs(@NonNull Context context) {
        this.context = context;
    }

    @NonNull
    public Sites readSites() {
        String sitesString = getDefaultPrefs().getString(KEY_SITES, "A;A;A;A;A");
        String[] words = sitesString.split(";");
        return new Sites(words[0], words[1], words[2], words[3], words[4]);
    }

    public void writeSites(@NonNull Sites sites) {
        String sitesString = sites.getFrench() + ";" + sites.getItalian() + ";"
                + sites.getSpanish() + ";" + sites.getEnglish() + ";" + sites.getDeutsch();
        SharedPreferences.Editor editor = getEditor();
        editor.putString(KEY_SITES, sitesString);
        editor.apply();
    }

    @Nullable
    public String readCookie() {
        return getDefaultPrefs().getString(KEY_COOKIE, null);
    }

    public void writeCookie(@Nullable String cookie) {
        SharedPreferences.Editor editor = getEditor();
        editor.putString(KEY_COOKIE, cookie);
        editor.apply();
    }

    public boolean isIntroShown() {
        return getDefaultPrefs().getBoolean(KEY_INTRO_SHOWN, false);
    }

    public void writeIntroShown(boolean introShown) {
        SharedPreferences.Editor editor = getEditor();
        editor.putBoolean(KEY_INTRO_SHOWN, introShown);
        editor.apply();
    }

    @NonNull
    private SharedPreferences getDefaultPrefs() {
        return context.getSharedPreferences(context.getPackageName(), Context.MODE_PRIVATE);
    }

    @NonNull
    private SharedPreferences.Editor getEditor() {
        return getDefaultPrefs().edit();
    }

}
```